msr_batch.c

```diff
@@ -226,7 +226,7 @@ void msrbatch_cleanup(void)
     if (cdev_created)
     {
         cdev_created = 0;
-        device_destroy(cdev_class, MKDEV(majordev, 0));
+        device_destroy(cdev_class, MKDEV(majordev_msr_batch, 0));
```

Comment:
Since you are already passing ```mdev``` in as a parameter to ```int msrbatch_init(int msdev)```, you could also pass the same parameter to ```msrbatch_cleanup()``` and would only need to keep the static ints that are defined in msr_entry.c and could remote the ```static int majordev_msr_batch``` variable.